script/update-content-versions-and-check-for-archived-slugs.ts

```diff
@@ -0,0 +1,105 @@
+import books from '../src/config.books.json';
+import createArchiveLoader from '../src/gateways/createArchiveLoader';
+import createOSWebLoader from '../src/gateways/createOSWebLoader';
+
+interface ArchivedSlug {
+  pathname: string;
+  bookId: string;
+  pageId: string;
+}
+
+type ArchivedSlugs = ArchivedSlug[];
+
+const { book, newVersion } = argv as any as {
+  book: string
+  newVersion: string | number;
+};
+
+(global as any).fetch = fetch;
+
+const booksPath = path.resolve(__dirname, '../src/config.books.json');
+const archivedSlugsPath = path.resolve(__dirname, '../src/archived-slugs.json');
+
+const archiveLoader = createArchiveLoader(`${ARCHIVE_URL}${REACT_APP_ARCHIVE_URL}`);
+const osWebLoader = createOSWebLoader(`${ARCHIVE_URL}${REACT_APP_OS_WEB_API_URL}`);
+
+async function processBooks() {
+  const booksData = Object.entries(books);
+  const bookToUpdateIndex = booksData.findIndex(([id]) => id === book);
+  const bookToUpdate = booksData[bookToUpdateIndex] as [string, { defaultVersion: string }] | undefined;
+
+  if (!bookToUpdate) {
+    console.error(`book ${book} not found`); // tslint:disable-line:no-console
+    return;
+  }
+
+  const [bookId, { defaultVersion }] = bookToUpdate;
+
+  if (defaultVersion === newVersion.toString()) {
+    console.log(`$${bookId} alredy at desired version.`); // tslint:disable-line:no-console
+    return;
+  }
+
+  const { title, tree: currentTree } = await archiveLoader.book(bookId, defaultVersion).load()
+    .catch((error) => {
+      // tslint:disable-next-line: no-console
+      console.log(`error while loading book ${bookId} with defaultVersion ${defaultVersion}`);
+      throw error;
+    });
+  const { tree: newTree } = await archiveLoader.book(bookId, newVersion.toString()).load()
+    .catch((error) => {
+      console.log(`error while loading book ${bookId} with newVersion ${newVersion}`); // tslint:disable-line:no-console
+      throw error;
+    });
+  const bookSlug = await osWebLoader.getBookSlugFromId(bookId)
+    .catch((error) => {
+      console.log(`error while loading slug for bookId ${bookId}`); // tslint:disable-line:no-console
+      throw error;
+    });
+
+  const flatCurrentTree = flattenArchiveTree(currentTree);
+  const flatNewTree = flattenArchiveTree(newTree);
+  const missingPages = [...archivedSlugs] as ArchivedSlugs;
+
+  const findArchivedSlug = (section: LinkedArchiveTreeNode) => (
+    { pageId, bookId: pageBookId, pathname }: ArchivedSlug
+  ) => pageId === section.id && pageBookId === bookId && pathname.split('/').pop() === section.slug;
+
+  const formatSection = (section: LinkedArchiveTreeNode) => ({
+    bookId,
+    pageId: section.id,
+    pathname: `/books/${bookSlug}/pages/${section.slug}`,
+  });
+
+  let countNewArchivedSlugs = 0;
+  for (const section of flatCurrentTree) {
+    if (
+      !flatNewTree.find((newSection) => newSection.slug === section.slug)
```

Comment:
instead of just finding slug by slug, try finding each page from the old version in the new version by id, then comparing the slugs